links/src/main/java/juja/microservices/links/service/impl/LinksServiceImpl.java

```diff
@@ -24,12 +24,19 @@
     @Override
     public Link saveLink(SaveLinkRequest request) {
         Link link;
+        String url = request.getUrl();
 
-        try {
-            link = linksRepository.getLinkByURL(request.getOwner(), request.getUrl());
-            log.info("Link already exists {}. ", link.toString());
-        } catch (NotFoundException ex) {
-            link = linksRepository.saveLink(request.getOwner(), request.getUrl());
+        link = linksRepository.getLinkByURL(url);
```

Comment:
In fact there were both "owner" field and that getLinkByURLAndOwner() before I started with this issue. I've renamed the old one and added the new getLinkByURL() method because there is no necessity to know who "owns" a link (one who saved it before) when you're trying to save a link, all you need to know - is it present in the database or not. Indeed, we receive a request with 2 fields - URL and ID of the user, which sends the request. Nothing about the "owner" in connection with URL (it can be anyone, not the user which's saving a link right now). 
BTW a few days ago I've proposed  to rename that request's "owner" to "user" but you argued so we've got this ambiguity when the same "owner" name is used to identify two different actors (one, who sends request and another one, who "owns" previously saved link).

Reply:
For instance:

User A save url 'www.com'
User B save url 'www.com'

Both users A and B hide these links. 

Then:

User B save url 'url.com' second time. What link have to be unhidden?
Mongo findOne find random link and it can be user's A link.

After that user A have active link 'www.com' and doesn't understand why.

Reply:
Great! My decision excludes multiple saves of the same link. It can be stored just once. And the person, who has saved it, is it's "owner", so the "owner" (or a moderator) only can hide the link (with getLinkByURLAndOwner() called in process - the reason I've kept it). If you're trying to save a link which is already present in DB and not hidden, you receive a warning. Or in the case it's present and hidden you unhide it making visible to all. This way after A has saved and made the link hidden, B can make it not hidden only. There are no issues with Mongo at all.